ui/src/components/MainView.tsx

```diff
@@ -8,8 +8,10 @@ import PartyListEdit from './PartyListEdit';
 
 const MainView: React.FC = () => {
   const party = useParty();
-  const myUser = useFetchByKey(User, () => party, [party]).contract?.payload;
-  const allUsers = useQuery(User, () => ({}), []).contracts.map((user) => user.payload);
+  const userHook = useFetchByKey(User, () => party, [party]);
```

Comment:
```suggestion
  const myUserResult = useFetchByKey(User, () => party, [party]);
```
The reis not a hook but the `useFetchByKey` function is the hook. Same below.

I'm wondering if the API for hooks is too clumsy here. Currently, `useFetchByKey` returns something of type `{ contract: CreateEvent<T> | null; loading: boolean }`. If it returned something of type `[CreateEvent<T> | null, boolean]`, we could write
```
const [myUserContract] = useFetchByKey(...)
```
instead and ignore the boolean we don't care about.

Reply:
That could be better, and would be consistent with `useExercise` and `useExerciseByKey`. We would need to remove both `QueryResult` and `FetchResult` if we do that.